extras/rya.indexing.pcj/src/main/java/org/apache/rya/indexing/pcj/storage/accumulo/AccumuloPeriodicQueryResultStorage.java

```diff
@@ -0,0 +1,271 @@
+        this.accumuloConn = accumuloConn;
+        this.ryaInstance = ryaInstance;
+        String user = accumuloConn.whoami();
+        try {
+            this.auths = accumuloConn.securityOperations().getUserAuthorizations(user);
+        } catch (AccumuloException | AccumuloSecurityException e) {
+            throw new RuntimeException("Unable access user: " + user + "authorizations.");
+        }
+    }
+
+    @Override
+    public String createPeriodicQuery(String sparql) throws PeriodicQueryStorageException {
+        Preconditions.checkNotNull(sparql);
+        String queryId = pcjIdFactory.nextId();
+        return createPeriodicQuery(queryId, sparql);
+    }
+    
+    @Override
+    public String createPeriodicQuery(String queryId, String sparql) throws PeriodicQueryStorageException {
+        Set<String> bindingNames;
+        try {
+            bindingNames = new AggregateVariableRemover().getNonAggregationVariables(sparql);
+        } catch (MalformedQueryException e) {
+            throw new PeriodicQueryStorageException(e.getMessage());
+        }
+        List<String> varOrderList = new ArrayList<>();
+        varOrderList.add(PeriodicQueryResultStorage.PeriodicBinId);
+        varOrderList.addAll(bindingNames);
+        createPeriodicQuery(queryId, sparql, new VariableOrder(varOrderList));
+        return queryId;
+    }
+
+    @Override
+    public void createPeriodicQuery(String queryId, String sparql, VariableOrder order) throws PeriodicQueryStorageException {
+        Preconditions.checkNotNull(sparql);
+        Preconditions.checkNotNull(queryId);
+        Preconditions.checkNotNull(order);
+        Preconditions.checkArgument(PeriodicQueryResultStorage.PeriodicBinId.equals(order.getVariableOrders().get(0)),
+                "periodicBinId binding name must occur first in VariableOrder.");
+        String tableName = tableNameFactory.makeTableName(ryaInstance, queryId);
+        Set<VariableOrder> varOrders = new HashSet<>();
+        varOrders.add(order);
+        try {
+            pcjTables.createPcjTable(accumuloConn, tableName, varOrders, sparql);
+        } catch (Exception e) {
+            throw new PeriodicQueryStorageException(e.getMessage());
+        }
+    }
+
+    @Override
+    public PeriodicQueryStorageMetadata getPeriodicQueryMetadata(String queryId) throws PeriodicQueryStorageException {
+        try {
+            return new PeriodicQueryStorageMetadata(
+                    pcjTables.getPcjMetadata(accumuloConn, tableNameFactory.makeTableName(ryaInstance, queryId)));
+        } catch (Exception e) {
+            throw new PeriodicQueryStorageException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void addPeriodicQueryResults(String queryId, Collection<VisibilityBindingSet> results) throws PeriodicQueryStorageException {
+        results.forEach(x -> Preconditions.checkArgument(x.hasBinding(PeriodicQueryResultStorage.PeriodicBinId),
```

Comment:
No.  The constructor takes in an Accumulo Connector and uses this each time it needs to retrieve results.